src/psyclone/tests/psyir/nodes/call_test.py

```diff
@@ -628,6 +630,560 @@ def test_call_get_callees_local(fortran_reader):
+  end subroutine
+
+end module some_mod'''
+
+    root_node: Node = fortran_reader.psyir_from_source(code)
+
+    routine_main: Routine = root_node.walk(Routine)[0]
+    assert routine_main.name == "main"
+
+    routine_match: Routine = root_node.walk(Routine)[1]
+    assert routine_match.name == "foo"
+
+    call_foo: Call = routine_main.walk(Call)[0]
+    assert call_foo.routine.name == "foo"
+
+    (result, arg_idx_list) = call_foo.get_callee()
+    result: Routine
+
+    assert len(arg_idx_list) == 3
+    assert arg_idx_list[0] == 2
+    assert arg_idx_list[1] == 0
+    assert arg_idx_list[2] == 1
+
+    assert result is routine_match
+
+
+def test_call_get_callee_5_optional_and_named_arguments(fortran_reader):
+    '''
+    Check that optional and named arguments have been correlated correctly
+    when the call is to a generic interface.
+    '''
+    code = '''
+module some_mod
+  implicit none
+contains
+
+  subroutine main()
+    integer :: e, f, g
+    call foo(b=e, a=f)
+  end subroutine
+
+  ! Matching routine
+  subroutine foo(a, b, c)
+    integer :: a, b
+    integer, optional :: c
+  end subroutine
+
+end module some_mod'''
+
+    root_node: Node = fortran_reader.psyir_from_source(code)
+
+    routine_main: Routine = root_node.walk(Routine)[0]
+    assert routine_main.name == "main"
+
+    routine_match: Routine = root_node.walk(Routine)[1]
+    assert routine_match.name == "foo"
+
+    call_foo: Call = routine_main.walk(Call)[0]
+    assert call_foo.routine.name == "foo"
+
+    (result, arg_idx_list) = call_foo.get_callee()
+    result: Routine
+
+    assert len(arg_idx_list) == 2
+    assert arg_idx_list[0] == 1
+    assert arg_idx_list[1] == 0
+
+    assert result is routine_match
+
+
+_code_test_get_callee_6 = '''
+module some_mod
+  implicit none
+
+  interface foo
+    procedure foo_a, foo_b, foo_c
+  end interface
+contains
+
+  subroutine main()
+    integer :: e_int, f_int, g_int
+    real :: e_real, f_real, g_real
+
+    ! Should match foo_a
+    call foo(e_int, f_int)
+
+    ! Should match foo_a
+    call foo(e_int, f_int, g_int)
+
+
+    ! Should match foo_b
+    call foo(e_real, f_int)
+
+    ! Should match foo_b
+    call foo(e_real, f_int, g_int)
+
+    ! Should match foo_b
+    call foo(e_real, c=f_int, b=g_int)
+
+
+    ! Should match foo_c
+    call foo(e_int, f_real, g_int)
+
+    ! Should match foo_c
+    call foo(b=e_real, a=f_int)
+
+    ! Should match foo_c
+    call foo(b=e_real, a=f_int, g_int)
+  end subroutine
+
+  subroutine foo_a(a, b, c)
+    integer :: a, b
+    integer, optional :: c
+  end subroutine
+
+  subroutine foo_b(a, b, c)
+    real :: a
+    integer :: b
+    integer, optional :: c
+  end subroutine
+
+  subroutine foo_c(a, b, c)
+    integer :: a
+    real :: b
+    integer, optional :: c
+  end subroutine
+
+end module some_mod'''
+
+
+def test_call_get_callee_6_interfaces_0_0(fortran_reader):
+    '''
+    Check that optional and named arguments have been correlated correctly
```

Comment:
Please extend each of the following docstrings to briefly say precisely which aspect of the correlation is being tested.

Reply:
OK. I did the same with all following ones as well and added also to the code which call is related to which test.